test/docker-tests.robot

```diff
@@ -133,6 +133,23 @@ Three node compose example
   Host count on port 7302 should be 3
   [Teardown]  Delete compose from  ../docker-compose/marklogic-cluster-centos.yaml
 
+Two node compose example with node joining Default group
```

Comment:
The most recent change has made the tests pass in Jenkins but the test no longer changes the group and makes sure the group has been changed so I'm not sure if compose test 6 covers the changes in this PR. We already had Default group working. What should be tested for this change would be switching the group to anything other than Default then making sure the group after the ML server has been spun up is correctly.

Reply:
As per the AC, the expectation is that host should join the group given in MARKLOGIC_GROUP env parameter whether it's Default, enode or dnode. I changed the test to verify node2 should join the group given in the new parameter with the value as the default group as in this story, we are just handling group parameter while adding the host to the cluster and not creating any additional groups in the cluster. Also, the compose example only creates the Default group. More tests can be added when we have the parameter to create groups while creating a bootstrap host.

Reply:
I understand the compose example only creates the default group, my review is that given the changes if we want to fully test them the compose test should create another group and test joining it. From my perspective this test just makes sure something we were already doing is working properly and doesn't make sure these new changes are working as intended.

Reply:
compose definition cannot create another group as there is no environment parameter available for that functionality.

Reply:
You could write the test in the robot framework and use API calls to the Marklogic server to create the groups. I'm sure there are also other alternatives, that's just what is coming to mind for me. If compose can't properly test this feature witch from what you are saying it sounds like it can't, we shouldn't use compose to write the test for this feature.

Reply:
Currently in the docker test framework, compose is used to test all the multi-node MarkLogic docker functionalities hence I used compose to test MarkLogic_Group env param while joining the cluster. I can add API calls to create groups and test to join a different group on the cluster.

Reply:
Added tests to verify node 2 joining group enode in a ML cluster.